### PR TITLE
feat: qbd direct export settings employee mapping and automap employe…

### PR DESCRIPTION
**Pull request by DhaaraniCIT**
… update

### Description
In QBD integration updated the export settings with the employee mapping and auto map employee sections. added the employee mapping with the reference from PRD

## Clickup
Please add link here

### CODE REVIEW

**Comment by DhaaraniCIT** [comment]
Screenshot 2024-09-25 at 3 15 06 PM

File: src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.ts [hunk 1]
```diff
@@ -50,6 +52,10 @@ export class QbdDirectExportSettingsComponent implements OnInit{
 
   splitExpenseGroupingOptions: QBDExportSettingFormOption[] = QbdDirectExportSettingModel.splitExpenseGroupingOptions();
 
+  autoMapEmployeeOptions: QBDExportSettingFormOption[] = QbdDirectExportSettingModel.autoMapEmployeeOptions();
```
Comment (ashwin1111) [inline src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.ts:55]: this is going to be a toggle

File: src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.ts [hunk 2]
```diff
@@ -77,6 +83,13 @@ export class QbdDirectExportSettingsComponent implements OnInit{
     private mappingService: MappingService
   ) { }
 
+  getEmployeeIsDisabled(): boolean {
```
Comment (ashwin1111) [inline src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.ts:86]: better name

File: src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.ts [hunk 3]
```diff
@@ -129,10 +142,22 @@ export class QbdDirectExportSettingsComponent implements OnInit{
     });
   }
 
+  employeeMappingWatcher() {
+    this.exportSettingsForm.controls.reimbursableExportType.valueChanges.subscribe((reimbursableExportTypeValue) => {
+      if (reimbursableExportTypeValue === QbdDirectReimbursableExpensesObject.BILL) {
+        this.exportSettingsForm.controls.employeeMapping.patchValue(EmployeeFieldMapping.VENDOR);
+      } else if (reimbursableExportTypeValue === QbdDirectReimbursableExpensesObject.CHECK) {
```
Comment (ashwin1111) [inline src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.ts:149]: not going to have cheque

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**